### ISSUE TITLE
User service not running with supplementary groups

### ISSUE THREAD

**Issue by wjmelements** [systemd/systemd#15659]
**245.5-2**

Distro: **Arch Linux**

I created a user service for a system (uid < 1000) user that belongs to group `systemd-journal`. The service reads from another user service using `journalctl --user-unit` and notifies me of categories of issues. I am doing it this way so that I can allow regular users to login as this user and administer its services without granting `wheel` to them or the system user.

However, the service did not work, giving the error `No journal files were opened due to insufficient permissions.` Upon investigating, it seems the result of `groups` inside the user service is only the user group, excluding `systemd-journal`. This matches the behavior documented in `man 5 systemd.exec`: "If no group is set, the default group of the user is used.". However, if I try to specify User, Group, or SupplementaryGroups, I get `Failed to determine supplementary groups: Operation not permitted` or `Changing group credentials failed: Operation not permitted`.

The workaround is that this particular service must be run as a system service that specifies User, Group, etc. I would prefer not to have to do this because I want regular users to be able to administer this service without `wheel` because they have the permission. I also suspect that if this was not a system user, I may not have this issue.

**Expected behaviour you didn't see**
 > User service for user in group `systemd-journal` can read from journal

**Unexpected behaviour you saw**
 > User service does not have `systemd-journal` group when running

**Steps to reproduce the problem**
1. Create system user with `useradd --system`.
2. Add system user to `systemd-journal`
3. Install the following user service and run it with `systemctl --user`
```
[Unit]
Description=systemd user groups

[Service]
ExecStart=/usr/bin/groups

[Install]
WantedBy=default.target
```
3. Check logs with `journalctl --user-unit`

**Comment by poettering** [comment]
so let me get this right. If you do `/cat/proc/$PID/status` of the `systemd --user` instance itself shows the group in question in the `Groups:` line, but then, when it spawns a child it's gone from the file of that process?

**Comment by wjmelements** [comment]
Yes. Here is the difference between the `systemctl --user` process and the root systemctl running as the user:

```
# root
$ cat /proc/267319/status | grep Groups
Groups:	100 190 
# user
$ cat /proc/306438/status | grep Groups
Groups:	991 
```

where 190 is `systemd-journal` and 991 is the user's personal gid.

**Comment by wjmelements** [comment]
The shell process that invokes `systemctl --user` gets all three groups:

```
# bash
$ cat /proc/307303/status | grep Group
Groups:	100 190 991 
```

**Comment by scratchy** [comment]
can confirm this bug..

```
#  grep '^Groups' /proc/2383857/status
Groups: 1 2 3 4 6 1001

# grep '^Groups' /proc/$$/status
Groups: 1 2 3 4 6 121 1001
```

Running a systemd service (java) and forking a subprocess (bash) out of it with missing group permissions. As described above already the parent is missing the group it seems...

did you find a workaround?!

**Comment by Waester** [comment]
I am also encountering the same bug mentioned above.

I have a user service which need access to /dev/dri/renderD128, which is granted by being part of the group "video".
Running `/usr/bin/groups` in the terminal shows that my user is part of it.

`10000 video sudoers`

running the same command from a `systemctl --user` service shows a different story.

`10000 sudoers`

**Comment by huksley** [comment]
I also experience this, the systemd user process running without docker group so have no access to docker.

**Comment by manuelstein** [comment]
I also experience this
